extension-manifest-v3/src/background/onboarding.js

```diff
@@ -12,32 +12,45 @@
 import { observe } from '/store/options.js';
 import { sendShowIframeMessage } from './utils/iframe.js';
 import { tabStats } from './stats.js';
+import { showOperaSerpNotification } from '/utils/opera-serp.js';
 
 const NOTIFICATION_DELAY = 24 * 60 * 60 * 1000; // a day in milliseconds
 const NOTIFICATION_TRACKERS_THRESHOLD = 10;
 
-let shownAt = undefined;
-observe('onboarding', (onboarding) => {
-  shownAt = !onboarding.done && onboarding.shownAt;
+let done = false;
+let shownAt = 0;
 
-  if (!onboarding.done && !onboarding.shownAt) {
+observe(null, (options) => {
+  ({ done, shownAt } = options.onboarding);
+
+  if (!done && !shownAt) {
     chrome.tabs.create({
       url: chrome.runtime.getURL('/pages/onboarding/index.html'),
     });
   }
 });
 
 chrome.webNavigation.onCompleted.addListener((details) => {
-  if (
-    details.frameId === 0 &&
-    shownAt &&
-    Date.now() - shownAt > NOTIFICATION_DELAY
-  ) {
-    setTimeout(() => {
-      const stats = tabStats.get(details.tabId);
-      if (stats && stats.trackers.length >= NOTIFICATION_TRACKERS_THRESHOLD) {
-        sendShowIframeMessage(details.tabId, 'pages/onboarding/iframe.html');
+  if (details.frameId === 0) {
+    if (!done) {
+      if (shownAt && Date.now() - shownAt > NOTIFICATION_DELAY) {
+        setTimeout(() => {
+          const stats = tabStats.get(details.tabId);
+          if (
+            stats &&
+            stats.trackers.length >= NOTIFICATION_TRACKERS_THRESHOLD
+          ) {
+            sendShowIframeMessage(
+              details.tabId,
+              'pages/onboarding/iframe.html',
+            );
+          }
+        }, 1000);
       }
-    }, 1000);
+    } else if (__PLATFORM__ === 'opera') {
+      showOperaSerpNotification(() => {
+        sendShowIframeMessage(details.tabId, 'pages/onboarding/serp.html');
```

Comment:
shouldn't this logic be encapsulated inside `showOperaSerpNotification` - magic sting suggests it is wrong place

Reply:
Do you mean `__PLATFORM__...` condition? It must be outside, so the bundler can remove this code (and all of the imports) from platforms, which don't need it.

Reply:
I've meant the page path `'pages/onboarding/serp.html'`

Reply:
the `__PLATFORM__` condition is fine to stay, I ask about it's contents